tree/tree/test/TChainParsing.cxx

```diff
@@ -184,6 +185,92 @@ TEST(TChainParsing, GlobbingWithNonExistingDir)
+
+   const auto *cwd = gSystem->WorkingDirectory();
+
+   ASSERT_EQ(gSystem->mkdir("testglob"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir1"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir1/subdir11"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir2"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir2/subdir21"), 0);
+   ASSERT_EQ(gSystem->mkdir("testglob/subdir3"), 0);
+
+   // Unsorted to also test sorting the final list of globbed files.
+   std::vector<std::string> fileNames{"0a.root",
+                                      "testglob/subdir1/1a.root",
+                                      "testglob/subdir1/1b.root",
+                                      "testglob/subdir3/3a.root",
+                                      "testglob/subdir2/subdir21/21a.root",
+                                      "testglob/subdir2/subdir21/21b.root",
+                                      "testglob/subdir1/subdir11/11b.root",
+                                      "testglob/subdir1/subdir11/11a.root"};
+   for (const auto &fileName : fileNames) {
+      FillTree(fileName.c_str(), "events");
+   }
+
+   TChain nodir, none, nested, globDir;
+
+   nodir.Add("*");
+   const auto *nodirFiles = nodir.GetListOfFiles();
+   ASSERT_TRUE(nodirFiles);
+   EXPECT_EQ(nodirFiles->GetEntries(), 1);
+   auto expectedFileNameNodir = gSystem->ConcatFileName(cwd, "0a.root");
+   EXPECT_STREQ(nodirFiles->At(0)->GetTitle(), gSystem->UnixPathName(expectedFileNameNodir));
+   delete[] expectedFileNameNodir;
+
+   none.Add("testglob/*");
+   const auto *noneChainFiles = none.GetListOfFiles();
+   ASSERT_TRUE(noneChainFiles);
+   EXPECT_EQ(noneChainFiles->GetEntries(), 0);
+
+   nested.Add("testglob/*/*/*.root");
+   const auto *nestedChainFiles = nested.GetListOfFiles();
+   ASSERT_TRUE(nestedChainFiles);
+   EXPECT_EQ(nestedChainFiles->GetEntries(), 4);
+   std::vector<std::string> expectedFileNamesNested{
+      "testglob/subdir1/subdir11/11a.root", "testglob/subdir1/subdir11/11b.root", "testglob/subdir2/subdir21/21a.root",
+      "testglob/subdir2/subdir21/21b.root"};
+   auto nestedChainFileNames = GetFileNamesVec(nestedChainFiles);
+   EXPECT_VEC_EQ(nestedChainFileNames, expectedFileNamesNested);
+
+   globDir.Add("*/subdir[0-9]/*");
+   const auto *globDirChainFiles = globDir.GetListOfFiles();
+   ASSERT_TRUE(globDirChainFiles);
+   EXPECT_EQ(globDirChainFiles->GetEntries(), 3);
+   std::vector<char *> expectedFileNamesGlobDir{gSystem->ConcatFileName(cwd, "testglob/subdir1/1a.root"),
+                                                gSystem->ConcatFileName(cwd, "testglob/subdir1/1b.root"),
+                                                gSystem->ConcatFileName(cwd, "testglob/subdir3/3a.root")};
+   auto globDirChainFileNames = GetFileNamesVec(globDirChainFiles);
+   for (std::size_t i = 0; i < expectedFileNamesGlobDir.size(); i++) {
+      EXPECT_EQ(globDirChainFileNames[i], gSystem->UnixPathName(expectedFileNamesGlobDir[i]));
```

Comment:
```suggestion
      EXPECT_EQ(globDirChainFileNames[i], std::string{gSystem->UnixPathName(expectedFileNamesGlobDir[i].c_str()}));
```
This looks like a logic error to me! Here, both arguments are currently `char *`, and if you compare them with `EXPECT_EQ` it compares the memory location, not the string content. If you follow my suggestion to use `std::string` where you can, this is avoided.